tests/components/google/test_init.py

```diff
@@ -305,38 +262,98 @@ async def test_invalid_calendar_yaml(
     # Integration fails to setup
     assert not await component_setup()
 
-    assert not hass.states.get("calendar.backyard_light")
+    assert not hass.states.get(TEST_YAML_ENTITY)
 
     mock_notification.assert_not_called()
 
 
+@pytest.mark.parametrize(
+    "google_config_track_new,calendars_config,expect_tracked",
+    [
+        (None, [], True),  # CONF_TRACK_NEW is unset
+        (True, [], True),
+        (False, [], False),
+    ],
+    ids=["default", "True", "False"],
+)
+async def test_track_new(
+    hass: HomeAssistant,
+    mock_token_read: None,
+    component_setup: ComponentSetup,
+    mock_calendars_list: ApiResult,
+    test_api_calendar: dict[str, Any],
+    mock_calendars_yaml: None,
+    expect_tracked: bool,
+) -> None:
+    """Test behavior of configuration.yaml settings for tracking new calendars not in the config."""
+
+    mock_calendars_list({"items": [test_api_calendar]})
+    assert await component_setup()
+
+    # The calendar does not
+    state = hass.states.get(TEST_API_ENTITY)
+    if expect_tracked:
```

Comment:
Can we pass a `State` instance as test parameter instead so we can avoid the conditional logic here?

Reply:
I did this, however, i was unable to set direct comparisons with `==` because the `context` fields in `State` are different between the actual state and expected state input used for the fixture. I still like the style of this approach, so I added a method that does the state comparisons handling the `None` case.  I searched through existing tests that create a State object but none of them were doing comparisons like this. Let me know if you have a better suggestion.

Reply:
I don't have a better suggestion now. Let's go with this for now. 👍